addons/binding/org.openhab.binding.zway/src/main/java/org/openhab/binding/zway/internal/discovery/ZWayBridgeDiscoveryService.java

```diff
@@ -0,0 +1,186 @@
+                            Short prefix = address.getNetworkPrefixLength();
+
+                            logger.debug("Scan IP address for Z-Way Server: {}", ipAddress);
+
+                            // Search on localhost first
+                            scheduler.execute(new ZWayServerScan(ipAddress));
+
+                            String subnet = ipAddress + "/" + prefix;
+                            SubnetUtils utils = new SubnetUtils(subnet);
+                            String[] addresses = utils.getInfo().getAllAddresses();
+
+                            for (String addressInSubnet : addresses) {
+                                scheduler.execute(new ZWayServerScan(addressInSubnet));
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (SocketException e) {
+            logger.warn("Error occured while searching Z-Way servers ({})", e.getMessage());
+        }
+    }
+
+    public boolean pingHost(String host, int port, int timeout) {
+        try (Socket socket = new Socket()) {
+            socket.connect(new InetSocketAddress(host, port), timeout);
+            return true;
+        } catch (IOException e) {
+            return false; // Either timeout or unreachable or failed DNS lookup.
+        }
+    }
+
+    public class ZWayServerScan implements Runnable {
+        private String ipAddress;
+
+        public ZWayServerScan(String ipAddress) {
+            this.ipAddress = ipAddress;
+        }
+
+        @Override
+        public void run() {
+            if (!pingHost(ipAddress, 8083, 500)) {
+                return; // Error occured while searching Z-Way servers (Unreachable)
+            }
+
+            try {
+                URL url = new URL("http://" + ipAddress + ":8083/ZAutomation/api/v1/status");
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+                if (connection.getResponseCode() == 401) {
+                    logger.info("(DiscoverySuccess) Z-Way server found on host: {}", ipAddress);
+
+                    ThingUID thingUID = new ThingUID(ZWayBindingConstants.THING_TYPE_BRIDGE,
+                            ipAddress.replaceAll("\\.", "_"));
+
+                    // Attention: if is already present as thing in the ThingRegistry
+                    // the configuration for thing will be updated!
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID)
+                            .withProperty(ZWayBindingConstants.BRIDGE_CONFIG_ZWAY_SERVER_IP_ADDRESS, ipAddress)
+                            .withLabel("Z-Way Server " + ipAddress).build();
+                    thingDiscovered(discoveryResult);
+                } else {
+                    logger.info("No Z-Way server found on host: {}", ipAddress);
+                }
+            } catch (Exception e) {
+                logger.info(e.getClass().toString());
```

Comment:
I guess this is some unexpected system state here? Better log a warning with the full stacktrace (i.e. `logger.warn("Discovery resulted in an unexpected exception", e)`.